cf-agent/files_select.c

```diff
@@ -494,6 +494,11 @@ static int SelectIsSymLinkTo(EvalContext *ctx, char *filename, Rlist *crit)
             Log(LOG_LEVEL_DEBUG, "Skipping readlink() on directory %s", filename);
             return false;
         }
+        else if (S_ISREG(statbuf.st_mode))
```

Comment:
I would rather change consolidate both of these into this:

```
else if (!S_ISLNK(statbuf.st_mode))
{
    Log(LOG_LEVEL_DEBUG, "Skipping readlink() on non-symlink %s", filename);
    return false;
}
```

Then you catch other error cases such as device files, sockets and fifos. We've been getting several fixes lately for these cases by @worr, for example.